viewer/packages/cad-model/src/wrappers/CogniteCadModel.ts

```diff
@@ -156,23 +160,25 @@ export class CogniteCadModel implements CdfModelNodeCollectionDataProvider {
    *
    * @param nodeCollection Dynamic set of nodes to apply the provided appearance to.
    * @param appearance Appearance to style the provided set with.
+   * @param importance The importance of this style. Can be used to manually order the order of styles, this can avoid the order of adding styles affecting the outcome. Optional and defaults to 0.
    * @example
    * ```js
    * model.setDefaultNodeAppearance({ rendererGhosted: true });
    * const visibleNodes = new TreeIndexNodeCollection(someTreeIndices);
    * model.assignStyledNodeCollection(visibleSet, { rendererGhosted: false });
    * ```
    */
-  assignStyledNodeCollection(nodeCollection: NodeCollection, appearance: NodeAppearance): void {
+  assignStyledNodeCollection(nodeCollection: NodeCollection, appearance: NodeAppearance, importance: number = 0): void {
     MetricsLogger.trackCadModelStyled(nodeCollection.classToken, appearance);
 
     const index = this._styledNodeCollections.findIndex(x => x.nodeCollection === nodeCollection);
     if (index !== -1) {
       this._styledNodeCollections[index].appearance = appearance;
     } else {
-      this._styledNodeCollections.push({ nodeCollection: nodeCollection, appearance });
+      this._styledNodeCollections.push({ nodeCollection: nodeCollection, appearance, importance });
     }
-    this.cadNode.nodeAppearanceProvider.assignStyledNodeCollection(nodeCollection, appearance);
+    this._styledNodeCollections.sort((a, b) => a.importance - b.importance);
```

Comment:
Javascript builtin sorting is not stable, meaning that if e.g. all collections have importance 0, we are not guaranteed that they are stored in the order they are added, which is important for our current behavior. Lodash's `sortBy` function should be stable